src/Plugin/EntityBrowser/Display/Modal.php

```diff
@@ -177,7 +173,7 @@ public function addAjax(array &$form) {
     );
 
     $form['actions']['submit']['#ajax'] = array(
-      'callback' => array(get_class($this), 'widgetAjaxCallback'),
+      'callback' => array($this, 'widgetAjaxCallback'),
```

Comment:
I assumed as much, but that callback is using $this->entities. That would need to be refactored first (and I'd be +1 to that, along the lines of the other pull request, using $form_state for the storage of that).

I just added that as a quickfix to test it using the modal again, not really related to this issue. Can remove it from the pull request if you prefer that, but modal is completely dead right now in HEAD :)